tests/keras_tests/feature_networks_tests/feature_networks/linear_collapsing_test.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 Sony Semiconductors Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from abc import ABC
+import model_compression_toolkit as mct
+import tensorflow as tf
+from tests.common_tests.helpers.generate_test_tp_model import generate_test_tp_model
+from model_compression_toolkit.tpc_models.keras_tp_models.keras_default import generate_keras_default_tpc
+from tests.keras_tests.feature_networks_tests.base_keras_feature_test import BaseKerasFeatureNetworkTest
+import numpy as np
+from tests.common_tests.helpers.tensors_compare import cosine_similarity
+
+keras = tf.keras
+layers = keras.layers
+tp = mct.target_platform
+
+
+class BaseConv2DCollapsingTest(BaseKerasFeatureNetworkTest, ABC):
+
+    def __init__(self, unit_test):
+        super(BaseConv2DCollapsingTest, self).__init__(unit_test=unit_test, input_shape=(32,32,16))
+
+    def get_tpc(self):
+        tp = generate_test_tp_model({'weights_n_bits': 32,
+                                     'activation_n_bits': 32,
+                                     'enable_weights_quantization': False,
+                                     'enable_activation_quantization': False})
+        return generate_keras_default_tpc(name="linear_collapsing_test", tp_model=tp)
+
+    def get_quantization_config(self):
+        return mct.QuantizationConfig(mct.QuantizationErrorMethod.NOCLIPPING, mct.QuantizationErrorMethod.NOCLIPPING,
+                                      False, False, True)
+
+    def compare(self, quantized_model, float_model, input_x=None, quantization_info=None):
+        y = float_model.predict(input_x)
+        y_hat = quantized_model.predict(input_x)
+        self.unit_test.assertTrue(y.shape == y_hat.shape, msg=f'out shape is not as expected!')
+        cs = cosine_similarity(y, y_hat)
+        self.unit_test.assertTrue(np.isclose(cs, 1), msg=f'fail cosine similarity check:{cs}')
+
```

Comment:
Let's check more expected things. For example number of conv layers in the "quantized" model, cases where bias wasn't originally in the first conv but was added due to collapsing...

Reply:
Done